test/squid_test.rb

```diff
@@ -0,0 +1,785 @@
+      end
+
+      it "saves the system service" do
+        expect(service).to receive(:save)
+
+        squid.Write
+      end
+
+      context "and the service is correctly saved" do
+        before do
+          allow(service).to receive(:save).and_return(true)
+        end
+
+        it "returns true" do
+          expect(squid.Write).to eq(true)
+        end
+      end
+
+      context "and the service is not correctly saved" do
+        before do
+          allow(service).to receive(:save).and_return(false)
+        end
+
+        it "returns false" do
+          expect(squid.Write).to eq(false)
+        end
+      end
+    end
+  end
+
+  describe ".readHttpPorts" do
+    before do
+      allow(Yast::SCR).to receive(:Read).with(path_matching(/\.squid\..*http_port/))
+        .and_return [["localhost:3128"], ["80", "transparent"]]
+    end
+
+    it "reads the ports in different supported formats" do
+      squid.readHttpPorts
+      expect(squid.http_ports).to eq [
+        {"host"=>"localhost", "port"=>"3128"},
+        {"host"=>"", "port"=>"80", "transparent"=>true}
+      ]
+    end
+  end
+
+  describe ".readHttpAccesses" do
+    before do
+      allow(Yast::SCR).to receive(:Read).with(path_matching(/\.squid\..*http_access/))
+        .and_return [
+          ["allow", "manager", "localhost"], ["deny", "manager"], ["deny", "!Safe_ports"],
+          ["deny", "CONNECT", "!SSL_ports"], ["allow", "localhost"],
+          ["allow", "localhost_public"], ["deny", "all"]
+        ]
+    end
+
+    it "reads entries in several formats" do
+      squid.readHttpAccesses
+      expect(squid.http_accesses).to eq [
+        { "acl" => ["manager", "localhost"],  "allow" => true},
+        { "acl" => ["manager"],               "allow" => false},
+        { "acl" => ["!Safe_ports"],           "allow" => false},
+        { "acl" => ["CONNECT", "!SSL_ports"], "allow" => false},
+        { "acl" => ["localhost"],             "allow" => true},
+        { "acl" => ["localhost_public"],      "allow" => true},
+        { "acl" => ["all"],                   "allow" => false}
+      ]
+    end
+  end
+
+  describe ".readRefreshPatterns" do
+    before do
+      allow(Yast::SCR).to receive(:Read).with(path_matching(/\.squid\..*refresh_pattern/))
+        .and_return [
+          ["^ftp:", "1440", "20%", "10080"],
+          ["-i", "^gopher:", "1440", "0%", "1440"],
+          [".", "0", "20%", "4320"]
+        ]
+    end
+
+    it "parses several kinds of patterns" do
+      squid.readRefreshPatterns
+      expect(squid.refresh_patterns).to eq [
+        {
+          "case_sensitive" => true, "max" => "10080", "min" => "1440",
+          "percent" => "20", "regexp"=>"^ftp:"
+        },
+        {
+          "case_sensitive" => false, "max" => "1440", "min" => "1440",
+          "percent" => "0", "regexp"=>"^gopher:"
+        },
+        {
+          "case_sensitive" => true, "max" => "4320", "min" => "0",
+          "percent" => "20", "regexp"=>"."
```

Comment:
For me both of the test cases ( this and previous ) will be more readable if pairs of data value and result value was close together, so I can see what changes in what.

So something like:
```
PATTERNS = {
  ["^ftp:", "1440", "20%", "10080"] => {
          "case_sensitive" => true, "max" => "10080", "min" => "1440",
          "percent" => "20", "regexp"=>"^ftp:"
        },
["-i", "^gopher:", "1440", "0%", "1440"] => {
          "case_sensitive" => false, "max" => "1440", "min" => "1440",
          "percent" => "0", "regexp"=>"^gopher:"
        },
...
}

      allow(Yast::SCR).to receive(:Read).with(path_matching(/\.squid\..*refresh_pattern/)).and_return(PATTERNS.keys)
      squid.readRefreshPatterns
      expect(squid.refresh_patterns).to eq PATTERNS.values
```

Reply:
Changed